### PR TITLE
WIP: kafka support, as well as streaming binlogs

### PR DESCRIPTION
**Pull request by osheroff**
@vanchi-zendesk not much to see yet but may as well code in public

### CODE REVIEW

**Comment by osheroff** [comment]
@vanchi-zendesk this is ready for review, but it didn't actually include kafka support; instead it isolated out booting up maxwell from the commandline, basic config file style stuff, and storing the current binlog position in a file.

File: src/main/java/com/zendesk/maxwell/BinlogPosition.java [hunk 1]
```diff
@@ -5,27 +5,22 @@
...
-			rs = c.createStatement().executeQuery("SHOW MASTER STATUS");
-			rs.next();
-            return new BinlogPosition(rs.getInt("Position"), rs.getString("File"));
-		} catch (SQLException e) {
-			// TODO be smart about permission denied here
-			return null;
-		}
+		rs = c.createStatement().executeQuery("SHOW MASTER STATUS");
+		rs.next();
+		return new BinlogPosition(rs.getInt("Position"), rs.getString("File"));
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/BinlogPosition.java]: here there's not much code change, but I'm throwing the error instead of papering over it.

**Comment by osheroff** [comment]
pingo

**Comment by vanchi-zendesk** [comment]
​Sorry, I will get to this tomorrow. Too much firefighting this week.​

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	public int    mysqlPort;
+	public String mysqlUser;
+	public String mysqlPassword;
+
+	public String currentPositionFile;
+
+	private BinlogPosition initialPosition;
+
+	public Connection getMasterConnection() throws SQLException {
+		return DriverManager.getConnection("jdbc:mysql://" + mysqlHost + ":" + mysqlPort, mysqlUser, mysqlPassword);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: This probably needs to be rewritten, (in my pr) ?

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+		File f = new File(this.currentPositionFile);
+		FileWriter fw = new FileWriter(f);
+		try {
+			p.store(fw, "");
+			this.initialPosition = position;
+		} finally {
+			fw.close();
+		}
+	}
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: I am surprised this is in a file... I thought you would read from mysql... Is this stop-gap?

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java [hunk 4]
```diff
@@ -34,24 +34,86 @@
...
 
 		TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
 		this.binlogEventListener = new MaxwellBinlogEventListener(queue);
+
+		this.replicator = new OpenReplicator();
+
+		this.replicator.setHost("127.0.0.1");
+		this.replicator.setUser("maxwell");
+		this.replicator.setPassword("maxwell");
+		this.replicator.setBinlogEventListener(this.binlogEventListener);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: I am guessing, these are defaults. If they are, port is missing.

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java [hunk 5]
```diff
@@ -117,143 +177,10 @@ private void processQueryEvent(QueryEvent event) throws SchemaSyncError {
...
-		r.registgerEventParser(new DeleteRowsEventV2Parser());
-		r.registgerEventParser(new FormatDescriptionEventParser());
-
-		return r;
-	}
-
-	private void initParser(String fileName, long position) throws Exception {
-		FileBasedBinlogParser bp = getDefaultBinlogParser();
-
-		bp.setStartPosition(position);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: This is no longer needed because we are directly connecting to tcp stream?

**Comment by vanchi-zendesk** [comment]
:+1:  Sorry for the delay.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: yup, eventually.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 3 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: haven't 100% decided where to store last-position.  probably mysql, but we might also just write the file/position into the kafka stream and pop the last-written off the stream to recover the position...

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java (hunk 4 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: I think these can go away.

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java (hunk 5 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: we're using the OpenReplicator's objects directly instead of implementing our own.

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: Ah ok. I was talking about setStartPosition() which is missing in the other code.. We set the FileName and FilePath in setBinlogPosition().

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: Ha, I am wrong. You are setting the offset up there. It is setBinlogFilePath which is missing now.

**Pull request merged**